sabakan/generator.go

```diff
@@ -195,20 +195,20 @@ func (g *Generator) getWorkerTmpl(role string) nodeTemplate {
 	panic("BUG: instantiating for invalid role: " + role)
 }
 
-// SetWaitSeconds set seconds before removing retiring nodes from the cluster.
+// SetWaitSeconds set seconds before removing retired nodes from the cluster.
 func (g *Generator) SetWaitSeconds(secs float64) {
 	g.waitSeconds = secs
 }
 
+// removeMachine removes the specified machine from machines slice.
+// If m is nil, this returns the same slice as input.
 func removeMachine(machines []*Machine, m *Machine) []*Machine {
-	var ret []*Machine
 	for i, mm := range machines {
 		if m == mm {
-			ret = append(machines[:i], machines[i+1:]...)
-			break
+			return append(machines[:i], machines[i+1:]...)
 		}
 	}
-	return ret
+	return machines
```

Comment:
I think `panic` is appropriate if `m` must be in `machines` .

Reply:
Thanks. I fixed so.